### PR TITLE
fix the collector fail to start issue after updated the new config deps

### PR DESCRIPTION
**Pull request by mxiamxia**
**Description:**
Fix the new config interface error and taken care of windows exit code(0) issue

### CODE REVIEW

**Review by pingleig — APPROVED** [review]
Can you remove the config.TryConfig as well (and its usage in config factory test).

File: cmd/awscollector/main_windows.go [hunk 1]
```diff
@@ -45,3 +47,11 @@ func runService(params service.Parameters) error {
 
 	return nil
 }
+
+// Has to set exit code 0 for the fatal errors when run the collector as service in Windows
```
Comment (pingleig) [inline cmd/awscollector/main_windows.go:51]: This is a new level of hack 😅 for #340

**Review by vastin — APPROVED** [review]
LGTM

File: cmd/awscollector/main_windows.go (hunk 1 above)
Comment (mxiamxia) [reply cmd/awscollector/main_windows.go:52]: any better idea? I think it seems better than previous one :)

**Comment by mxiamxia** [comment]
> Can you remove the config.TryConfig as well (and its usage in config factory test).

done, thx

File: cmd/awscollector/main_windows.go (hunk 1 above)
Comment (pingleig) [reply cmd/awscollector/main_windows.go:52]: yeah, it's better than previous one (which did even work ...

**Pull request merged**